zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/distribution/CommandDistributionIdempotencyTest.java

```diff
@@ -622,22 +626,36 @@ public void test() {
 
   private static void interceptAcknowledgement(final Scenario scenario) {
     final var hasInterceptedAlready = new AtomicBoolean(false);
-    final var acknowledgementCounter = new AtomicInteger(0);
+    final ConcurrentHashMap<Long, UnifiedRecordValue> commands = new ConcurrentHashMap<>();
+
+    // This callback will be called for:
+    // 1. the effective command on the target partition (e.g. 2)
+    // 2. the acknowledgements on the command distributing initiating partition (e.g. 1)
     ENGINE.interceptInterPartitionCommands(
-        (receiverPartitionId, valueType, intent, recordKey, command) -> {
-          if (intent != CommandDistributionIntent.ACKNOWLEDGE) {
+        (receiverPartitionId, valueType, intent, recordKey, commandValue) -> {
+
+          // store the original command for our scenario (which is received on the target partition)
+          if (valueType == scenario.valueType && intent == scenario.intent) {
+            commands.put(recordKey, commandValue);
             return true;
           }
 
-          final var currentAcknowledgementCount = acknowledgementCounter.incrementAndGet();
-
-          if (hasInterceptedAlready.get()
-              || currentAcknowledgementCount < scenario.expectedAmountOfDistributions) {
```

Comment:
Please carefully check if the new logic is equivalent. I assume here we are skipping a bunch of unrelated acknowledgements (`scenario.expectedAmountOfDistributions`) and then swallow to first relevant acknowledgement to trigger the "retry mechanism" 🕵️